File: frontend/src/types/session.ts
export type UserType = "superadmin" | "user";
export type MembershipRole = "club_admin" | "club_staff" | "member";
export type MembershipStatus = "active" | "invited" | "suspended" | "inactive";
export type RoleShell = "admin" | "player" | "superadmin" | null;
export type SessionMenuShell = "admin" | "player" | "superadmin";

export interface SessionUser {
  id: string;
  email: string;
  display_name: string;
  user_type: UserType;
}

export interface AvailableClub {
  club_id: string;
  club_name: string;
  club_slug: string;
  membership_role: MembershipRole | null;
  membership_status: MembershipStatus | null;
  selectable: boolean;
  is_primary_hint: boolean;
}

export interface SelectedClub {
  id: string;
  name: string;
  slug: string;
  location: string;
  timezone: string;
  branding: {
    logo_object_key: string | null;
    name: string | null;
  };
}

export interface SessionMenuItem {
  key: string;
  label: string;
  path: string;
  shell: SessionMenuShell;
  domain: string;
  module_key: string | null;
}

export interface SessionBootstrap {
  user: SessionUser;
  available_clubs: AvailableClub[];
  selected_club_id: string | null;
  selected_club: SelectedClub | null;
  club_selection_required: boolean;
  role_shell: RoleShell;
  default_workspace: string | null;
  landing_path: string;
  module_flags: Record<string, boolean>;
  menu_items?: SessionMenuItem[];
  permissions: string[];
  feature_flags: Record<string, boolean>;
}

export interface TokenResponse {
  access_token: string;
  token_type: "bearer";
  expires_in_seconds: number;
  user: SessionUser;
}

File: backend/app/schemas/session.py
from __future__ import annotations

import uuid
from typing import Literal

from pydantic import BaseModel, ConfigDict

from app.models import ClubMembershipRole, ClubMembershipStatus, UserType


class AvailableClubSummary(BaseModel):
    club_id: uuid.UUID
    club_name: str
    club_slug: str
    membership_role: ClubMembershipRole | None
    membership_status: ClubMembershipStatus | None
    selectable: bool
    is_primary_hint: bool


class SelectedClubSummary(BaseModel):
    id: uuid.UUID
    name: str
    slug: str
    timezone: str
    branding: dict[str, str | None]


class SessionUserSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    email: str
    display_name: str
    user_type: UserType


class SessionBootstrapResponse(BaseModel):
    user: SessionUserSummary
    available_clubs: list[AvailableClubSummary]
    selected_club_id: uuid.UUID | None
    selected_club: SelectedClubSummary | None
    club_selection_required: bool
    role_shell: Literal["admin", "player"] | None
    default_workspace: str | None
    landing_path: str
    module_flags: dict[str, bool]
    permissions: list[str]
    feature_flags: dict[str, bool]
